### PR TITLE
Feature/sort

### PR DESCRIPTION
**Pull request by makitango**
Some additions:
- based on recent input, I also integrated a suggested unit conversion feature to switch between inches and centimeters.
- liters as total volume (and thus the most relevant data) has been included in the list display

### CODE REVIEW

File: components/AirlineList/index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                  {`${Math.round(convertUnits(cabinBagWidth, unitSystem))} `}
+                  {" x "}
+                  {`${Math.round(convertUnits(cabinBagHeight, unitSystem))} ${
+                    unitSystem === "metric" ? "cm" : "in"
+                  }`}
+                  {" | "}
+                  {Math.round(
+                    (cabinBagLength * cabinBagWidth * cabinBagHeight) / 1000
+                  )}{" "}
+                  l
```
Comment (doemser) [inline components/AirlineList/index.js:61]: Calling functions within your jsx like this is at least wild and definitely super hard to follow.

In React, it's generally okay to call functions in JSX, but it's important to remember that these functions will be called on every render. If the function is computationally expensive or causes side effects, it could negatively impact the performance of your app or create unexpected behavior.

These functions are pretty lightweight so this should not be a problem, but it's  no method you should get used to.

**Review by doemser — CHANGES_REQUESTED** [review]
Nice that you solved this challenging tasks. Unfortunately the logic you came up with is hard to follow and potentially brings performance issues, I left you a comment on that.

I would suggest that you refactor your logic, as it will stand in your way and slow you down on further iterations. You can best fix this now. 

I made a prototype for you that isolates the problem and should help you to find a better path:

https://codesandbox.io/s/immutable-waterfall-d8jsfy?file=/src/App.js

As this is a pretty big chunk of your programs logic I expect you to ***deeply understand*** code before you take from that prototype. If you use chunks of it please comment on it with explanations, I will check on your understanding for this ;) 

This is about your apps architecture, do not worry on spending too much time on this, the better you layed out your app, the easier you will have it on every other feature

**Comment by makitango** [comment]
> Nice that you solved this challenging tasks. Unfortunately the logic you came up with is hard to follow and potentially brings performance issues, I left you a comment on that.
> 
> I would suggest that you refactor your logic, as it will stand in your way and slow you down on further iterations. You can best fix this now.
> 
> I made a prototype for you that isolates the problem and should help you to find a better path:
> 
> https://codesandbox.io/s/immutable-waterfall-d8jsfy?file=/src/App.js
> 
> As this is a pretty big chunk of your programs logic I expect you to _**deeply understand**_ code before you take from that prototype. If you use chunks of it please comment on it with explanations, I will check on your understanding for this ;)
> 
> This is about your apps architecture, do not worry on spending too much time on this, the better you layed out your app, the easier you will have it on every other feature

I will start looking into it now, thank you for your major effort to demonstrate the issue. ;)

File: components/AirlineList/index.js (hunk 1 above)
Comment (makitango) [reply components/AirlineList/index.js:61]: > Calling functions within your jsx like this is at least wild and definitely super hard to follow.
> 
> In React, it's generally okay to call functions in JSX, but it's important to remember that these functions will be called on every render. If the function is computationally expensive or causes side effects, it could negatively impact the performance of your app or create unexpected behavior.
> 
> These functions are pretty lightweight so this should not be a problem, but it's no method you should get used to.

Should I instead better make variables for the caluclations and import them from a dedicated place, similar to components? I would also like to simplify the structure of the code and I fully agree, the more I execute in the main components, the harder it is to tell from the code how the site is supposed to look.

File: components/AirlineList/index.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  convertDimension,
+} from "@/utils";
+
+export default function AirlineList({ sortedAirlines, unitSystem }) {
+  return (
+    <>
+      {sortedAirlines.map(
+        ({ id, name, personalItem, cabinBag, freeCabinBag }) => {
+          return (
+            <div key={id} style={{ border: "1px solid black", padding: 10 }}>
```
Comment (djfarly) [inline components/AirlineList/index.js:13]: Please use styled components instead of inline styles

File: pages/_app.js [hunk 3]
```diff
@@ -1,12 +1,87 @@
...
+  const [unitSystem, setUnitSystem] = useState("metric");
+  const [sortedAirlines, setSortedAirlines] = useState(airlines);
+
+  function handleUnitSystemChange(option) {
+    if (option === "metric") {
+      setUnitSystem("metric");
+    } else if (option === "imperial") {
+      setUnitSystem("imperial");
+    }
+  }
```
Comment (djfarly) [inline pages/_app.js:16]: since option is either "metric" or "imperial" you can pass it to the setUnitSystem direcly
```suggestion
  function handleUnitSystemChange(option) {
    setUnitSystem(option);
  }
```

File: pages/_app.js [hunk 4]
```diff
@@ -1,12 +1,87 @@
 import GlobalStyle from "../styles";
-import { bags } from "../components/List/data";
-import List from "./List";
+import { airlines } from "../lib/data";
```
Comment (djfarly) [inline pages/_app.js:2]: Naming suggestion:
```suggestion
import { airlines as initialAirlines } from "../lib/data";
```

File: pages/_app.js [hunk 5]
```diff
@@ -1,12 +1,87 @@
...
+import { useState } from "react";
+import { calculateVolume } from "@/utils";
 
 export default function App({ Component, pageProps }) {
+  const [unitSystem, setUnitSystem] = useState("metric");
+  const [sortedAirlines, setSortedAirlines] = useState(airlines);
```
Comment (djfarly) [inline pages/_app.js:8]: calling the inital data initialAirlines frees up the name, so that the state can be called `airlines` which reads nicer in all of the following code imho.

```suggestion
  const [airlines, setAirlines] = useState(initialAirlines);
```

Of course you have to change `sortedAirlines` and `setSortedAirlines` everywhere

File: components/AirlineList/index.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                  <strong>{calculateVolume(cabinBag)} l</strong>
+                </p>
+                <p>
+                  <strong>
+                    Combined volume{" "}
+                    {calculateVolumeCombined(personalItem, cabinBag)}
+                    {" l"}
+                  </strong>
+                </p>
+                <p> {freeCabinBag ? "" : "💰 Cabin bag costs extra 💰"}</p>
```
Comment (djfarly) [inline components/AirlineList/index.js:46]: You can get rid of the full paragraph by wrapping the ternary around the element:
```suggestion
                {freeCabinBag ? "" : <p>💰 Cabin bag costs extra 💰</p>}
```

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Dominik,

this is the home stretch I promise! 🤞

Some more notes from my side!

File: components/AirlineList/index.js (hunk 2 above)
Comment (makitango) [reply components/AirlineList/index.js:13]: > Please use styled components instead of inline styles

Can I use this shortcut in the meantime until I come to the task of styling? I will use StyledComponents at the time, I promise. Just wanted to keep the snippet from Dominik because it made things easier to read in the list.

File: pages/_app.js (hunk 3 above)
Comment (makitango) [reply pages/_app.js:16]: Makes totally sense, applied.

File: pages/_app.js (hunk 4 above)
Comment (makitango) [reply pages/_app.js:2]: Uhm, this is weird. The imports listed are quite outdated, I currently have these:
`import GlobalStyle from "../styles";
import { airlines } from "../lib/data";
import { useState } from "react";
import { calculateVolume } from "@/utils";`

But I will apply the suggestion alongside the follow-up since it makes perfectly sense.

File: components/AirlineList/index.js (hunk 6 above)
Comment (makitango) [reply components/AirlineList/index.js:46]: Ingenious.

File: pages/_app.js (hunk 5 above)
Comment (makitango) [reply pages/_app.js:8]: Done as well.

File: components/AirlineList/index.js (hunk 2 above)
Comment (djfarly) [reply components/AirlineList/index.js]: ✅

**Review by djfarly — APPROVED** [review]
👍
